exercises/concept/bird-watcher/bird_watcher_test.go

```diff
@@ -0,0 +1,121 @@
+)
+
+func TestTotalBirdCount(t *testing.T) {
+	tests := []struct {
+		name       string
+		birdCounts []int
+		want       int
+	}{
+		{
+			name:       "calculates the correct total number of birds",
+			birdCounts: []int{9, 0, 8, 4, 5, 1, 3},
+			want:       30,
+		},
+		{
+			name:       "works for a short bird count list",
+			birdCounts: []int{2},
+			want:       2,
+		},
+		{
+			name:       "works for a long bird count list",
+			birdCounts: []int{2, 8, 4, 1, 3, 5, 0, 4, 1, 6, 0, 3, 0, 1, 5, 4, 1, 1, 2, 6},
+			want:       57,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := TotalBirdCount(tt.birdCounts); got != tt.want {
+				t.Errorf("TotalBirdCount(%v) = %v; want %v", tt.birdCounts, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestBirdsInWeek(t *testing.T) {
+	tests := []struct {
+		name       string
+		birdCounts []int
+		week       int
+		want       int
+	}{
+		{
+			name:       "calculates the number of birds in the first week",
+			birdCounts: []int{3, 0, 5, 1, 0, 4, 1, 0, 3, 4, 3, 0, 8, 0},
+			week:       1,
+			want:       14,
+		},
+		{
+			name:       "calculates the number of birds for a week in the middle of the log",
+			birdCounts: []int{4, 7, 3, 2, 1, 1, 2, 0, 2, 3, 2, 7, 1, 3, 0, 6, 5, 3, 7, 2, 3},
+			week:       2,
+			want:       18,
+		},
+		{
+			name:       "works when there is only one week",
+			birdCounts: []int{3, 0, 3, 3, 2, 1, 0},
+			week:       1,
+			want:       12,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := BirdsInWeek(tt.birdCounts, tt.week); got != tt.want {
+				t.Errorf("BirdsInWeek(%v) = %v; want %v", tt.birdCounts, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestFixBirdCount(t *testing.T) {
+	tests := []struct {
+		name       string
+		birdCounts []int
+		week       int
+		want       []int
+	}{
+		{
+			name:       "returns a bird count list with the corrected values",
+			birdCounts: []int{3, 0, 5, 1, 0, 4, 1, 0, 3, 4, 3, 0},
+			want:       []int{4, 0, 6, 1, 1, 4, 2, 0, 4, 4, 4, 0},
+		},
+		{
+			name:       "works for a short bird count list",
+			birdCounts: []int{4, 2},
+			want:       []int{5, 2},
+		},
+		{
+			name:       "works for a long bird count list",
+			birdCounts: []int{2, 8, 4, 1, 3, 5, 0, 4, 1, 6, 0, 3, 0, 1, 5, 4, 1, 1, 2, 6},
+			want:       []int{3, 8, 5, 1, 4, 5, 1, 4, 2, 6, 1, 3, 1, 1, 6, 4, 2, 1, 3, 6},
+		},
+		{
+			name:       "does not create new slice",
+			birdCounts: []int{3, 0, 5, 1, 0, 4, 1, 0, 3, 4, 3, 0},
+			want:       []int{4, 0, 6, 1, 1, 4, 2, 0, 4, 4, 4, 0},
+		},
+	}
+
+	// Test normal cases - last test needs to be tested differently
+	for _, tt := range tests[:len(tests)-1] {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := FixBirdCountLog(tt.birdCounts); !reflect.DeepEqual(tt.want, got) {
+				t.Errorf("FixBirdCountLog(%v) = %v; want %v", tt.birdCounts, got, tt.want)
+			}
+		})
+	}
+
+	// Test special case
+	t.Run("does not create new slice", func(t *testing.T) {
```

Comment:
I am not sure the test runner can handle this extra subtest, I will check later

Reply:
not sure why this was resolved, I checked it and indeed, the test code extraction cannot handle this separate subtest, the user will see the full code of the whole TestFixBirdCount test which is very confusing

Since I think fixing the test runner would take quite some time, I would recommend to move this subtest into a separate (non-sub) test TestFixBirdCount_DoesNotCreateNewSlice.

Reply:
Sorry, this was me marking it as resolved. I thought you were going to check after the merge.

Anyway, I made the change now.

Reply:
Ah ok, no, I don't like to merge it without proper test runner code extraction because then usually we get confused students opening issues.

I looked at the latest version you pushed. The subtest inside `TestFixBirdCountDoesNotCreateNewSlice` is obsolete now, there is only this one and you already have the name in the name of the parent test.

Reply:
Just removed the subtest.